sdk/go/common/util/gitutil/git_test.go

```diff
@@ -342,3 +382,108 @@ func TestTryGetVCSInfoFromSSHRemote(t *testing.T) {
 		assert.Equal(t, test.WantVCSInfo, got)
 	}
 }
+
+// mockSSHConfig allows tests to mock SSH key paths.
+type mockSSHConfig struct {
+	path string
+	err  error
+}
+
+// GetKeyPath returns a canned response for SSH config.
+func (c *mockSSHConfig) GetStrict(host, key string) (string, error) {
+	return c.path, c.err
+}
+
+func TestParseAuthURL(t *testing.T) {
+	t.Parallel()
+
+	//nolint: gosec
+	rand := rand.New(rand.NewSource(0))
+
+	generateSSHKey := func(t *testing.T, passphrase string) string {
+		// TODO(go1.21): clear(sshKeys)
+		for m := range sshKeys {
+			delete(sshKeys, m)
+		}
```

Comment:
Not a fan of having the global registry and having to clear it each test, but I can live with it for now.
In the future, we should probably turn parseAuthURL into something like:

```go
type authURLParser struct {
	sshConfig sshUserSettings
	sshKeys   sync.Map  // string => transport.AuthMethod
}

var defaultAuthURLParser = &authURLParser{
	sshConfig: ssh_config.DefaultUserSettings,
}

func parseAuthURL(url string) (...) {
	return defaultAuthURLParser.Parse(url)
}

func (p *authURLParser) Parse(url string) (...)
```

This will give tests a pristine environment to work with.